shared/ui/drop-down-menu/menu/menu.tsx

```diff
@@ -0,0 +1,40 @@
+import { ComponentPropsWithoutRef, ElementRef, ReactNode, forwardRef } from 'react';
+
+import * as RadixDropDown from '@radix-ui/react-dropdown-menu';
+import clsx from 'clsx';
+
+import s from './menu.module.scss';
+
+type MenuProps = {
+  modal?: boolean;
+  onOpenChange?: (open: boolean) => void;
+  open?: boolean;
+  portal?: boolean;
+  trigger: ReactNode;
+} & Omit<ComponentPropsWithoutRef<typeof RadixDropDown.Content>, 'asChild'>;
+
+export const Menu = forwardRef<ElementRef<typeof RadixDropDown.Content>, MenuProps>(
+  ({ children, className, modal, onOpenChange, open, portal = true, trigger, ...rest }, ref) => {
+    const menuContent = (
+      <RadixDropDown.Content
+        className={clsx(s.content, className)}
+        ref={ref}
+        {...rest}
+        onPointerDownOutside={e => {
+          if (!portal) {
+            e.detail.originalEvent.preventDefault();
```

Comment:
Why is it necessary to prevent the default action in case the dropdown menu does not use `Portal`? What breaks in this case?

Reply:
Using a portal is particularly useful for dropdown menus because it avoids common issues related to overflow, z-index, and event handling. The portal prop allows you to toggle this behavior, giving flexibility depending on the specific requirements of UI.
- If portal is true: <RadixDropDown.Portal> is used to wrap the menuContent, rendering it at a different location in the DOM tree, typically near the root.
- If portal is false: menuContent is rendered in place, without using a portal.

Reply:
Okay, thanks for clarifying the purpose of `Portal'. But going back to my question, why is it necessary to prevent the default browser action? Unless there is an explicit reason to do so, doing so "just in case" is bad practice.